rustfmt-core/rustfmt-lib/src/visitor.rs

```diff
@@ -121,6 +121,21 @@ impl<'b, 'a: 'b> FmtVisitor<'a> {
         let is_all_semicolons =
             |snippet: &str| snippet.chars().all(|c| c.is_whitespace() || c == ';');
         if is_all_semicolons(&self.snippet(stmt.span())) {
```

Comment:
IIRC there was an change upstream in rustc that added a new StmtKind variant that represented an empty `;` (without a tuple). Not necessarily part of this PR, but if you are interested, it may be worth seeing if the AST now represents multiple semis (`let foo = 1;;;;;`) in such a way that we can handle more easily

Reply:
good idea! fixed.